engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/multiinstance/MultiInstanceActivityTest.java

```diff
@@ -1013,6 +1013,63 @@ public void shouldSetLoopCounterVariable() {
             tuple(elementInstanceKeys.get(2), "3"));
   }
 
+  @Test
+  public void shouldCompleteBodyWhenCompleteConditionAccessInnerVariableEvaluateTrue() {
```

Comment:
🔧  The test verifies that `numberOfInstances` can be used in the `completionCondition` expression and that it has the correct value. I think your test case is fine, but it should be reflected in the test's method name.

❌ We also need a similar test for `numberOfActiveInstances`. It should make sure the value is correct, perhaps we should check it gets updated as well. I'm also curious what value it should have when the child instance completes. Will it be 0 or 1 in the case of a sequential multi-instance?

Reply:
> Will it be 0 or 1 in the case of a sequential multi-instance?

Hi @korthout . Currently I don't find a definition for this situation in the bpmn specification, I wonder if we should be consistent with the `camunda-bpm-platform` behavior? From looking at the [technical documentation](https://docs.camunda.org/manual/latest/reference/bpmn20/tasks/task-markers/) and manual testing, the conclusion is that `numberOfActiveInstances` is always 1 when the child instance completes.

<img width="951" alt="image" src="https://user-images.githubusercontent.com/24605947/158022927-2fe18aec-0619-4982-bfec-bac2e10d68dc.png">

WDYT? Hope to give me a conclusion after internal discussion. 🙇

Reply:
@Lzgabel While we're discussing this topic internally, please just use the approach used in Camunda 7. However, note that I may come back at this with the request to change it.